pkg/render/guardian.go

```diff
@@ -202,6 +200,8 @@ func (c *GuardianComponent) clusterRoleBinding() client.Object {
 }
 
 func (c *GuardianComponent) deployment() client.Object {
+	var replicas int32 = DefaultReplicas
```

Comment:
I'm not sure it makes sense to have a global DefaultReplicas variable - is there really a conceptual global default that applies to all deployments we might want?

Like I said above, this defaulting should probably be done in the controller code anyway.